### PR TITLE
Support IBM AIX

### PR DESCRIPTION
**Pull request by exmatrikulator**

### CODE REVIEW

File: conans/client/conf/__init__.py [hunk 1]
```diff
@@ -24,7 +24,8 @@
     SunOS:
     Arduino:
         board: ANY
-arch: [x86, x86_64, ppc64le, ppc64, armv6, armv7, armv7hf, armv8, sparc, sparcv9, mips, mips64, avr]
+    AIX:
+arch: [x86, x86_64, ppc64le, ppc64, armv6, armv7, armv7hf, armv8, sparc, sparcv9, mips, mips64, avr, ppc64]
```
Comment (lasote) [inline conans/client/conf/__init__.py]: ppc64 is already there

File: conans/client/detect.py [hunk 2]
```diff
@@ -212,7 +212,8 @@ def _detect_os_arch(result, output):
                      'i86pc': 'x86',
                      'amd64': 'x86_64',
                      'aarch64': 'armv8',
-                     'sun4v': 'sparc'}
+                     'sun4v': 'sparc',
+                     '00f768d14c00': 'ppc64'}
```
Comment (lasote) [inline conans/client/detect.py]: `platform.machine()` returns `00f768d14c00`? looks very random

File: conans/client/conf/__init__.py (hunk 1 above)
Comment (exmatrikulator) [reply conans/client/conf/__init__.py]: oh dam, that's correct ;)

File: conans/client/detect.py (hunk 2 above)
Comment (exmatrikulator) [reply conans/client/detect.py]: That's the return value.
Maybe python platform needs to be fixed first?

Comment (memsharded) [reply conans/client/detect.py]: Fixing python platform would take lot of time, and wouldn't be backwards compatible with many python versions people are already using, so we would need a workaround here.

I would say that ``00f768d14c00`` is some magic or random number of those machines, but maybe it is not the same in other AIX machines. In platforms with uname, ``uname -m`` should give the same result. Is that number the result, always in different AIX machines?

**Comment by memsharded** [comment]
Hi @exmatrikulator 

Why closing this PR? No need to use AIX anymore? Adding AIX as a setting to settings.yml can be done, and the detect stage is not critical, as the defaults can be always set by hand. 

Please tell. Also, if you added AIX just for you, you might be interested in the functionality we are working now for next release, that allows to automatically share and synchronize settings and other configurations easily: https://github.com/conan-io/conan/pull/1705

Thanks!

**Comment by exmatrikulator** [comment]
Oh sorry.
The pull request is only an Issue now, not anymore a pull request.
It seems, that is only working for me.
A more general solution would be better.

But I'm still interested in AIX support. ;)

**Comment by memsharded** [comment]
Ok, so for now:
- You can add AIX manually to your ``<userhome>/.conan/settings.yml`` file. That will work.
- No worry about the autodetect, even if it fails, you can just go to your default profile ``<userhome>/.conan/profiles/default`` and add AIX there too.

So the next steps would be to check if the build helpers, compiler configurations, etc., work there too. For example the ``CMake`` helper might have some internal checks for ``self.settings.os`` or for ``self.settings.compiler`` that won't work for other unknown compilers, lets say if you have a ``aix-gcc`` compiler. You might be good not using the ``CMake`` helper (or the ``ConfigureBuildEnvironment`` for makefiles/autotools), but directly hardcoding your build commands inside the ``build()`` method. And when you learn about how to make it work, we could help to integrate that in the conan codebase if possible.

I know nothing about AIX, and don't have access to one, but if you are working in that environment, there is no problem in adding support for it in conan, as it can be useful for other users, though we would need your contributions for that.
Many thanks!

**Pull request closed without merging**